src/gui/SideBarWidget.cpp

```diff
@@ -77,44 +77,49 @@ void SideBarWidget::paintEvent( QPaintEvent * )
 	p.drawPixmap( 2, 2, m_icon.transformed( QTransform().rotate( -90 ) ) );
 }
 
-
-
 void SideBarWidget::resizeEvent( QResizeEvent * )
 {
 	const int MARGIN = 6;
 	m_contents->setGeometry( MARGIN, 40 + MARGIN, width() - MARGIN * 2,
 						height() - MARGIN * 2 - 40 );
 	m_closeBtn->move(m_contents->geometry().width() - MARGIN - 5, 5);
+
+	if(!filterWidgetLayout) return;
+
+	if(width() < 300) {
```

Comment:
This hard-coded size still leads to overlaps on my machine if the width is only slightly larger than 300 pixels. Please note that a rendered text can use up different pixel widths depending on the font that's used and the DPI of the screen.

I wonder if it would be better to have a layout where the check boxes for "User Content" and "Factory Content" are in one line and the "Hidden Content" check box always in a line beneath them. Separating them somehow would make sense anyway as they are concerned with different things.

Additionally it could be ensured that the width of the tab is always wide enough to render the elements without overlaps or cutoffs.

Reply:
Okay I'll try a layout like that. I was just a bit concerned that the height of the checkbox area would take away from being able to actually see the folders

Reply:
So I've done this and it now looks like this for me by default
![Screenshot from 2024-06-16 17-19-30](https://github.com/LMMS/lmms/assets/62200778/867c9e9a-ccdd-456c-a173-1f9307b7c7dd)

Reply:
Thanks! It's still possible to resize the sidebar so that text is cut off. However, I think this was already the case before this pull request.

While checking the code I also noticed that it might make sense at some point to reimplement the `SideBarWidget` with layouts instead of it's own custom painting code and resize events. However, that's not in the scope of this pull request.